### PR TITLE
Performance improvement & autocomplete 

### PR DESCRIPTION
**Pull request by henryyang42**
- Changed Ngram size to reasonable range, index time is less than 90sec now.
- Removed ``courses = Course.objects.filter(pk__in=[c.pk for c in courses])`` as it is insanely slow.
- Fixed tokenizer regex from ``\w`` to ``a-zA-Z0-9_``.
- Added basic autocomplete feature as the screenshot shows.

Feature plan
- Enhance autocomplete feature & performance.
- Change tooltip position as it takes too much place and may hide autocomplete result sometimes.

### CODE REVIEW

File: search/views.py [hunk 1]
```diff
@@ -85,19 +81,39 @@ def search(request):
...
         courses_page = paginator.page(paginator.num_pages)
 
-    courses_list = courses_page.object_list. \
-        values('id', 'no', 'eng_title', 'chi_title', 'note', 'objective',
-               'time', 'time_token', 'teacher', 'room', 'credit',
-               'prerequisite', 'ge', 'code', 'ys')
+    course_list = [x.object.__dict__ for x in courses_page.object_list]
+    for x in course_list:
+        for key in [k for k in x.keys() if k.startswith('_')]:
+            x.pop(key, None)
```
Comment (afq984) [inline search/views.py]: I'm not sure if modifying the ``__dict__`` is ok, I'd say this:
```python
course_list = [
    {k: v for (k, v) in x.object.__dict__.items() if not k.startswith('_')}
    for x in courses_page.object_list
]
```

**Review by afq984 — COMMENTED** [review]
Reviewed

File: search/views.py (hunk 1 above)
Comment (henryyang42) [reply search/views.py]: Fixed.
Thanks for the reminder.

Comment (afq984) [reply search/views.py]: no problem
I've not test the code though

**Pull request merged**